PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt

```diff
@@ -58,31 +58,49 @@ class PayPalWebCheckoutClient internal constructor(
         checkoutOrderId = request.orderId
         analytics.notify(CheckoutEvent.STARTED, checkoutOrderId)
 
-        val tokenResult = authenticationSecureTokenServiceAPI.createLowScopedAccessToken()
-        return when (tokenResult) {
-            is CreateLowScopedAccessTokenResult.Failure -> {
-                analytics.notify(CheckoutEvent.AUTH_CHALLENGE_PRESENTATION_FAILED, checkoutOrderId)
-                PayPalPresentAuthChallengeResult.Failure(tokenResult.error)
+        val result = if (request.appSwitchWhenEligible) {
+            // Check app switch eligibility
+            val patchCcoResult = patchCCOWithAppSwitchEligibility(
+                context = activity,
+                orderId = request.orderId,
+                tokenType = TokenType.ORDER_ID,
+                merchantOptInForAppSwitch = request.appSwitchWhenEligible,
+                paypalNativeAppInstalled = true // TODO: implement native app installed check
+            )
+            // Get RedirectUrl
+            val launchUrl = when (patchCcoResult) {
+                is APIResult.Success -> patchCcoResult.data.launchUrl
+                is APIResult.Failure -> null
```

Comment:
```suggestion
                is APIResult.Failure -> buildFallbackURL()
```
🧶 : We could build the fallback URL here to make the implementation cleaner.

Reply:
ended cleaning up more, moved logic of buildUrl to `PayPalWebCheckoutClient` from `PayPalWebLauncher`
let me know what u think